### PR TITLE
[MIG][website_snippet_anchor] Migrate to v10

### PR DESCRIPTION
**Pull request by yajo**
All my own commits from v8 were squashed, module relicensed to LGPL, renamed to have `web_editor` prefix instead of `website`. [Relevant commit](https://github.com/OCA/website/pull/361/commits/f2b7987acf88527df2be301c3dd8ac94aa443c27).

@Tecnativa @dreambits

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Make the PR on OpenUpgrade for the renaming then.

**Comment by yajo** [comment]
The script is included, is it not needed here then?

**Comment by pedrobaeza** [comment]
You need to put it on base module, because you need a "neutral ground" where declaring the renaming, or Odoo will say: I don't have `website_snippet_anchor` module. For pure renamings, you have to put it here: https://github.com/OCA/OpenUpgrade/blob/10.0/odoo/addons/openupgrade_records/lib/apriori.py. For module merging operations, you have to put it on https://github.com/OCA/OpenUpgrade/blob/833bcee6670132ac410ab53dd0ffaa45ded1f43f/odoo/addons/base/migrations/10.0.1.3/pre-migration.py#L56

**Review by chienandalu — COMMENTED** [review]
@Yajo Couldn't make a functional review. Seems like runbot can't load the js:
image

**Comment by pedrobaeza** [comment]
@Yajo please check Travis status and make the migration script on OpenUpgrade repo.

**Comment by yajo** [comment]
I hope this update makes it work, and the migration script is now in https://github.com/OCA/OpenUpgrade/issues/964

File: web_editor_snippet_anchor/templates/assets.xml
Comment (MiquelRForgeFlow) [inline web_editor_snippet_anchor/templates/assets.xml]: web_editor

File: web_editor_snippet_anchor/templates/assets.xml
Comment (MiquelRForgeFlow) [inline web_editor_snippet_anchor/templates/assets.xml]: web_editor

File: web_editor_snippet_anchor/static/src/js/link_dialog.js
Comment (MiquelRForgeFlow) [inline web_editor_snippet_anchor/static/src/js/link_dialog.js]: web_editor

File: web_editor_snippet_anchor/static/src/js/link_dialog.js
Comment (MiquelRForgeFlow) [inline web_editor_snippet_anchor/static/src/js/link_dialog.js]: web_editor

File: web_editor_snippet_anchor/static/src/js/anchor_option.js
Comment (MiquelRForgeFlow) [inline web_editor_snippet_anchor/static/src/js/anchor_option.js]: web_editor

File: web_editor_snippet_anchor/i18n/af.po
Comment (mreficent) [inline web_editor_snippet_anchor/i18n/af.po]: web_editor

File: web_editor_snippet_anchor/i18n/af.po
Comment (MiquelRForgeFlow) [inline web_editor_snippet_anchor/i18n/af.po]: web_editor

File: web_editor_snippet_anchor/static/src/js/anchor_option.js
Comment (MiquelRForgeFlow) [inline web_editor_snippet_anchor/static/src/js/anchor_option.js]: the module now doesn't depend on website

**Review by MiquelRForgeFlow — CHANGES_REQUESTED** [review]
I think you should check the javascript files to make them compatible with the module

changing the .po files is optional  😜

**Comment by yajo** [comment]
Very good catch @mreficent, I reverted the rename because this actually depends on website. Besides, not much sense of this outside a website environment.

**Comment by pedrobaeza** [comment]
I'm afraid it's still red

File: website_snippet_anchor/templates/snippets.xml
Comment (mreficent) [inline website_snippet_anchor/templates/snippets.xml]: website

File: website_snippet_anchor/templates/assets.xml
Comment (mreficent) [inline website_snippet_anchor/templates/assets.xml]: website

File: website_snippet_anchor/static/src/xml/link_dialog.xml
Comment (mreficent) [inline website_snippet_anchor/static/src/xml/link_dialog.xml]: website

File: website_snippet_anchor/static/src/js/anchor_option.js
Comment (mreficent) [inline website_snippet_anchor/static/src/js/anchor_option.js]: website?

File: website_snippet_anchor/static/src/js/link_dialog.js
Comment (mreficent) [inline website_snippet_anchor/static/src/js/link_dialog.js]: website?

**Comment by yajo** [comment]
@mreficent Sorry, none of your comments apply now.

@pedrobaeza There seems to be a problem installing sass, or compass, or bootstrap-sass, or I don't know what. @moylop260 could you check please here too?

**Comment by moylop260** [comment]
I have using [investigating-travis-test-failures](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#investigating-travis-test-failures) trick and the error that I found is *Cannot load compass. Maybe you should install the compass gem using this extra argument: $ sudo gem install compass --pre*:

```log

2017-07-18 20:05:13,175 2859 INFO moylop260 odoo.tests.common: phantom_run executing phantomjs /.repo_requirements/odoo/odoo/tests/phantomtest.js {"code": "odo
o.__DEBUG__.services['web_tour.tour'].run('website_snippet_country_dropdown_tour_demo_page')", "db": "moylop260", "session_id": "b93bc2d44ceb9e7a98a4081bc679cc
8c520675a9", "timeout": 60, "ready": "odoo.__DEBUG__.services['web_tour.tour'].tours.website_snippet_country_dropdown_tour_demo_page.ready", "url_path": "/", "
port": 18069}
2017-07-18 20:05:13,277 2859 INFO moylop260 odoo.addons.base.ir.ir_http: Generating routing map
2017-07-18 20:05:14,607 2859 WARNING moylop260 odoo.addons.base.ir.ir_qweb.assetsbundle: ERROR: Cannot load compass.
Maybe you should install the compass gem using this extra argument:

    $ sudo gem install compass --pre
This error occured while compiling the bundle 'web.assets_frontend' containing:
    - /web/static/lib/bootstrap/less/variables.less
    - /web/static/lib/bootstrap/less/mixins/vendor-prefixes.less
    - /web/static/lib/bootstrap/less/mixins/buttons.less
    - /web/static/src/less/variables.less
    - /web/static/src/less/utils.less
    - /web_editor/static/src/less/web_editor.variables.less
    - /web/static/src/less/import_bootstrap.less
    - /web_editor/static/src/less/web_editor.common.less
    - /website/static/src/less/website.ui.less
    - /website/static/src/less/website.ui.components.less
    - /website/static/src/less/website.wrapwrap.less
    - /website/static/src/less/website.snippets.less
    - /website/static/src/less/website.zoomodoo.less
    - /website_portal/static/src/less/website_portal.less
    - /website_blog/static/src/less/website_blog.less
    - /website_snippet_country_dropdown/static/src/css/style.sass
    - /website_form/static/src/less/website_form.less
```


I ran that command but the issue is not fixed.

@lasley Any clues?

**Comment by moylop260** [comment]
FYI current stable 10.0 branch is failing too:

Translation's build:
 - https://travis-ci.org/OCA/website/jobs/253901140 (3 days ago) 🔴 

Previous build:
 - https://travis-ci.org/OCA/website/builds/252246305 (8 days ago) 🍏 

The MQT changes was 12 days ago so is discard it.

<s>Maybe a odoo change breaking this library?</s> Forget it, I used a commit sha -8 days ago and the error persist.

**Comment by yajo** [comment]
💚 😊

**Review by MiquelRForgeFlow — APPROVED** [review]

**Review by chienandalu — APPROVED** [review]
Finally tested on runbot! :+1:

**Review by moylop260 — APPROVED** [review]
Shouldn't squash fixup commits? (because is a migration)

**Comment by yajo** [comment]
Of course, I was waiting for approval before squashing, It's ready to merge now. Please somebody hit the button, it's ugly if I do it myself 😋

**Comment by pedrobaeza** [comment]
Let CIs finish

**Comment by pedrobaeza** [comment]
Travis is giving error now

**Comment by yajo** [comment]
Travis always tries surprises me.

- [Linter fails with `ImportError: No module named lxml`](https://travis-ci.org/OCA/website/jobs/255555596#L814).
- [`website_legal_page` test fails with 500 error](https://travis-ci.org/OCA/website/jobs/255555599#L1517). Hard to know who's to blame since current branch was broken due to https://github.com/OCA/maintainer-quality-tools/pull/465, which is fixed here but now reveals more problems.

Sorry to bother you again @moylop260 and @lasley, could you help here? It was green before the rebase...

**Comment by lasley** [comment]
Worth mentioning - tests are only failing on `odoo/odoo`

**Comment by moylop260** [comment]
@Yajo FYI yesterday I noticed the `ImportError lxml` error but is a travis internal error.
More info about:
 - https://github.com/travis-ci/travis-ci/issues/8132 and https://github.com/OCA/maintainer-quality-tools/issues/467

I saw that just the first build has this error. (Maybe Travis Testing?)

**Comment by moylop260** [comment]
Build lint error is unrelated because my last comment above.
Build odoo/odoo was error because a change of odoo or a flaky error but I rebuilt and now is green.

Using runbot I tested just this module and don't show errors:
 - screen shot 2017-07-20 at 13 49 57

So I will merge it

**Pull request merged**
